tracer/test/test-applications/integrations/Samples.InstrumentedTests/Propagation/String/StringJoinTests.cs

```diff
@@ -526,4 +526,48 @@ public void GivenStringJoinTGenericClass_WhenJoin_ResultIsOk()
             string.Join<ClassForStringTest>("|", list),
             () => string.Join<ClassForStringTest>("|", list));
     }
+
+#if NET9_0_OR_GREATER
+
+    [Fact]
+    public void GivenStringAndStringReadOnlySpan_WhenJoin_ResultIsOk()
+    {
+        var values = new string[] { TaintedString, UntaintedString, OtherTaintedString, OtherUntaintedString };
+        var span = new ReadOnlySpan<string>(values);
+        AssertTaintedFormatWithOriginalCallCheck(":+-TaintedString-+:|UntaintedString|:+-OtherTaintedString-+:|OtherUntaintedString",
+            System.String.Join("|", span),
+            () => System.String.Join("|", values)); // Use values here as we can not use span in the lambda expression
+    }
+
+    [Fact]
+    public void GivenATaintedStringInAStringReadOnlySpan_WhenCallingJoin_ResultIsTainted()
+    {
+        var values = new string[] { taintedValue, taintedValue};
+        var span = new ReadOnlySpan<string>(values);
+        AssertTaintedFormatWithOriginalCallCheck(":+-tainted-+:a:+-tainted-+:",
+            System.String.Join('a', span),
+            () => System.String.Join('a', values)); // Use values here as we can not use span in the lambda expression
```

Comment:
> // Use values here as we can not use span in the lambda expression

So are we _actually_ testing the code we think we are? 🤔 The lambda is for calculating the "expected" value right? If so then this is fine 👍

Reply:
That's the case.